python/tidy/servo_tidy/tidy.py

```diff
@@ -302,6 +307,44 @@ def check_toml(file_name, lines):
             yield (idx + 1, "found asterisk instead of minimum version number")
 
 
+def check_shell(file_name, lines):
+    if not file_name.endswith(".sh"):
+        raise StopIteration
+
+    shebang = "#!/usr/bin/env bash"
+    required_options = {"set -o errexit", "set -o nounset", "set -o pipefail"}
+    # A flag to indicate whether we've found a "real" statement in the script, e.g. not one of `set -o <option>`
+    found_script_statement = False
+
+    for idx, line in enumerate(lines):
+        if idx == 0:
```

Comment:
I think it would be cleaner to check the first line outside (before) the loop; that takes away a level of nesting inside the `for` and means we don't need `enumerate` anymore.